src/java.base/share/classes/javax/security/auth/Subject.java

```diff
@@ -319,6 +329,181 @@ public Subject run() {
         });
     }
 
+
+    private static enum SubjectStorage {
+        ACC, THREAD_LOCAL, SCOPE_LOCAL
+    };
+
+    private static final SubjectStorage SUBJECT_STORAGE
+            = SharedSecrets.getJavaLangAccess().systemAllowSecurityManager()
+                    ? SubjectStorage.ACC
+                    : SubjectStorage.THREAD_LOCAL;
+
+//    private static final ScopeLocal<Subject> SUBJECT_SCOPE
+//            = ScopeLocal.inheritableForType(Subject.class);
+
+    private static final InheritableThreadLocal<Subject> SUBJECT_THREAD_LOCAL =
+            new InheritableThreadLocal<Subject>() {
+                @Override protected Subject initialValue() {
+                    return null;
+                }
+            };
+
+    /**
+     * Return the current installed subject.
+     * <p>
+     * The current installed subject is installed by the {@link #call}
```

Comment:
s/The current installed subject/The current subject/

I am not sure "installed" is the best term to use, as I equate that most commonly with installing software on your disk. Maybe "set" or "registered"?

Reply:
I thought about "associated" but cannot precisely describe what it it is associated to. The "install" word actually comes from "install a SecurityManager with the `System::setSecurityManager` method".

We will need to find a word anyway. Only saying "current object" is also OK but too thin.